src/tests/Microsoft.PowerFx.Core.Tests.Shared/ListFunctionVisitorTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests
+{
+    public class ListFunctionVisitorTests : PowerFxTest
+    {
+        [Theory]
+        [InlineData("Abs(1)", "Abs", null)]
+        [InlineData("Abs(Abs(Abs(Abs(Abs(1)))))", "Abs", null)]
+        [InlineData("With({x:Today()}, x+1)", "With,Today", null)]
+        [InlineData("SomeNameSpace.Foo() + SomeNameSpace.Bar()", "Foo,Bar", "SomeNameSpace.Foo,SomeNameSpace.Bar")]
+        [InlineData("true And true", "", null)]
+        [InlineData("If(true, Blank(),Error())", "If,Blank,Error", null)]
+        public void ListFunctionNamesTest(string expression, string expectedNames, string expectedFullNames)
+        {
+            foreach (var textFirst in new bool[] { false, true })
+            {
+                if (textFirst)
```

Comment:
>textFirst [](http://example.com/codeflow?start=20&length=9)

It would be good to have a test for testFirst without the leading `=` sign. For example: expression = `Hello ${Sum(1,Sqrt(2))} world`, and check that the visitor returns `Sum,Sqrt`.